### PR TITLE
Date & Time Property Editor

### PR DESCRIPTION
**Pull request by warrenbuckley**

### CODE REVIEW

**Comment by warrenbuckley** [comment]
@nielslyngsoe or @madsrasmussen do you mind doing a review of this so far please

File: src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.test.ts [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+  it('should show a type=date field if the format only contains a date', async () => {
+    element.config = [{alias: 'format', value: 'YYYY-MM-dd'}];
+    await element.updateComplete;
+    expect(inputElement.type).to.equal('date');
+  });
+
+  it('should show a type=time field if the format only contains a time', async () => {
+    element.config = [{alias: 'format', value: 'HH:mm'}];
+    await element.updateComplete;
+    expect(inputElement.type).to.equal('time');
```
[the remark this answers is not in the document]
Comment (warrenbuckley) [reply src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.test.ts:38]: The offset is not implemented at all @nielslyngsoe

File: src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.element.ts [hunk 2]
```diff
@@ -10,14 +14,71 @@ import { UmbLitElement } from '@umbraco-cms/element';
...
+
+		// Based on the type of format string change the UUI-input type
+		const timeFormatPattern = /^h{1,2}:m{1,2}(:s{1,2})?\s?a?$/gmi;
+        if (this._format?.toLowerCase().match(timeFormatPattern)) {
+			this._inputType = "time";
+		}
+
+		// TODO: Warren - could this be handled in the C# server side code?
+		// When this is set to true then you need to check the Umbraco.Sys.ServerVariables that comes from C# server in a global JS obejct
+		// How do we handle this ?
```
[the remark this answers is not in the document]
Comment (warrenbuckley) [reply src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.element.ts:73]: Updated comment - as the offset part of this property editor has not been dealt with

File: src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.element.ts [hunk 3]
```diff
@@ -10,14 +14,71 @@ import { UmbLitElement } from '@umbraco-cms/element';
...
+		} else {
+			this._value = undefined;
+			this._valueString = undefined;
+		}
+	}
+
+	get value(): string | undefined {
+		if (this._value) {
+			const d = this._value;
+			return `${d.getFullYear()}-${d.getMonth() + 1}-${d.getDate()} ${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}`;
```
[the remark this answers is not in the document]
Comment (warrenbuckley) [reply src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.element.ts:35]: Not sure I fully understand what I need to do or fix up here @nielslyngsoe ?

File: src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nielslyngsoe) [reply src/backoffice/shared/property-editors/uis/date-picker/property-editor-ui-date-picker.test.ts:38]: I see, good with the comment. I will approve, then we most return to the case of offset time.

**Review by nielslyngsoe — APPROVED** [review]

**Pull request merged**